platform/troubleshoot/tsnet-connectivity.mdx

```diff
@@ -0,0 +1,190 @@
+
+```
+ERROR ts-net-controller tsnet/tsnet.go:148 Check if TSNet is online { 
+  "component": "vcluster", 
+  "failCounter": 5, 
+  "error": "vcluster tsnet server is not online: ... Tailscale could not connect to any relay server. ... The last login error was: register request: Post \"https://vcluster-test.io/coordinator/machine/register\": connection attempts aborted by context: context deadline exceeded"
+}
+```
+
+## Issue
+
+When the TSNet connection fails, you might observe the following:
+
+- **vCluster status discrepancy**: The vCluster shows as `Running` in the remote host cluster but is not marked as ready in the vCluster Platform UI.
+
+- **Successful local connection**: You can successfully connect to the vCluster using `vcluster connect` command.
+
+- **Platform communication blocked**: The vCluster cannot communicate with the platform for status updates and management operations.
+
+To verify TSNet connection issues, check the vCluster logs:
+
+```bash
+kubectl -n <namespace> logs <vcluster-pod> | grep -i "tsnet\|ts-net"
+```
+
+Look for errors related to connection timeouts or relay server failures.
+
+
+## Causes
+
+TSNet connection failures might occur due to the following:
+
+- **Restricted egress policies**: Host clusters with strict network policies (common in GKE, EKS, and other managed Kubernetes services) might block outbound connections to the coordination server.
+
+- **Ingress controller interference**: Ingress controllers like Istio might block or interfere with WebSocket upgrades required for TSNet communication.
+
+- **Direct connection attempts**: TSNet defaults to attempting direct peer-to-peer connections, which often fail in cloud environments with network restrictions.
+
+- **Firewall restrictions**: Corporate firewall rules or cloud security groups might block the required ports and protocols for TSNet communication.
+
+
+## Solution
+
+To resolve TSNet connection failures, disable direct connection mode and configure TSNet to use the platform agent for communication. This approach works better in restricted network environments.
+
+<Flow id="Resolve TSNet connection failures">
+
+<Step title="Configure TSNet to disable direct connections">
+
+Configure TSNet to disable direct connections
+
+Update your vCluster configuration to explicitly disable direct peer-to-peer connections:
+
+```yaml title="vcluster.yaml"
+controlPlane:
+  statefulSet:
+    env:
+      - name: "TS_DEBUG_DIAL_DIRECT"
+        value: "false"
+```
+</Step>
+
+<Step title="Apply the configuration">
+
+Update your existing vCluster with the new configuration:
+
+```bash
+vcluster create <vcluster-name> -f vcluster.yaml --upgrade
+```
+
+Or if using Helm directly:
+
+```bash
+helm upgrade <vcluster-name> vcluster --repo https://charts.loft.sh -f vcluster.yaml -n <namespace>
+```
+</Step>
+
+<Step title="Restart the vCluster pods">
+
+Force a restart of the vCluster pods to ensure the new environment variable takes effect:
```

Comment:
is it necessary? I think after running helm upgrade the vCluster pod should have restarted

Reply:
I usually check the age of the pod to make sure it was restarted and the changes was applied, a better way of verifying it would be checking the `TS_DEBUG_DIAL_DIRECT` variable is present in the pod spec